### ISSUE TITLE
Wrong table name is preventing clean uninstall

### ISSUE THREAD

**Issue by efanning** [catalyst/moodle-auth_saml2#212]
When I try to install this plugin into my Moodle server, it gets stuck on "Validating".  I have tried from Moodle's plugin site and uploading the zip file and get the same results.  I did have this installed at one point but had to uninstall because I tried upgrading to the 20180226 release and it started then.  I uninstalled and tried to install fresh to see if that fixes the issue, but it has not.  

I have turned on debugging and the screenshot below.  Is there something I'm missing?

![saml2_error](https://user-images.githubusercontent.com/35379527/37044673-11c8ccf8-2132-11e8-89c4-4896ca87d954.png)

**Comment by efanning** [comment]
This can be closed.  After turning on debugging to post this message, I searched the error message and increased the timeout from 30 to 120 and it is now installed.  Thank you for a great plugin.

**Comment by efanning** [comment]
I guess I spoke too soon, when I update Moodle database, I get the following error message.

```
Debug info: Table 'mdl_auth_samltwo_kvstore' already exists
CREATE TABLE mdl_auth_samltwo_kvstore (
id BIGINT(10) NOT NULL auto_increment,
type VARCHAR(30) CHARACTER SET utf8 COLLATE utf8_unicode_ci NOT NULL DEFAULT '',
k VARCHAR(50) CHARACTER SET utf8 COLLATE utf8_unicode_ci NOT NULL DEFAULT '',
value LONGTEXT CHARACTER SET utf8 COLLATE utf8_unicode_ci NOT NULL,
expire BIGINT(10),
CONSTRAINT PRIMARY KEY (id)
, UNIQUE KEY mdl_authsamlkvst_ktyp2_uix (k, type)
)
ENGINE = InnoDB
DEFAULT CHARACTER SET utf8
DEFAULT COLLATE = utf8_unicode_ci 
COMMENT='A key value store for SAML2'
Error code: ddlexecuteerror


×Stack trace:
line 492 of /lib/dml/moodle_database.php: ddl_change_structure_exception thrown
line 1041 of /lib/dml/mysqli_native_moodle_database.php: call to moodle_database->query_end()
line 77 of /lib/ddl/database_manager.php: call to mysqli_native_moodle_database->change_database_structure()
line 425 of /lib/ddl/database_manager.php: call to database_manager->execute_sql_arr()
line 370 of /lib/ddl/database_manager.php: call to database_manager->install_from_xmldb_structure()
line 590 of /lib/upgradelib.php: call to database_manager->install_from_xmldb_file()
line 1850 of /lib/upgradelib.php: call to upgrade_plugins()
line 694 of /admin/index.php: call to upgrade_noncore()
```

**Comment by danmarsden** [comment]
looks like that table doesn't get deleted on the uninstall process - I'm guessing because of the name "two" instead of "2" - you might need to delete the old table manually "mdl_auth_samltwo_kvstor" and then trigger the re-install again.

@roperto - we might need to rename that table to auth_saml2_kvstore and I think that may fix the uninstall process.

**Comment by efanning** [comment]
Ok, I'm not an expert at modifying database tables, can you assist?  I'm using Ubuntu 16.04 with mySQL.

**Comment by danmarsden** [comment]
If you are managing your own Moodle site this is something you should probably learn... :-)

A google search on how to drop a table in Mysql on ubuntu should help. you will want to drop the table "mdl_auth_samltwo_kvstore" - make sure you don't drop anything else though.

**Comment by efanning** [comment]
That's my plan.  Thank you, I started see what Google had for me when I posted the previous post, but got distracted on another issue.  Thank you for your assistance.  I'll let you know if that worked.

**Comment by roperto** [comment]
Hi @danmarsden , it seems like a few tests are disabled in the CI, including the plugin validation which should have caught that issue. I created the issue above to address that, I just need to get some time allocated here so I can work on those issues.

**Comment by efanning** [comment]
I was successful at getting the table removed manually and the "Upgrade Moodle Database" completed successfully.  Thank you for your time.

**Comment by roperto** [comment]
Just fixed this in #213 , I'll close this when merged
